muxer.go

```diff
@@ -0,0 +1,411 @@
+package astits
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/asticode/go-astikit"
+	"io"
+)
+
+const (
+	StartPID           uint16 = 0x0100
+	PMTStartPID        uint16 = 0x1000
+	ProgramNumberStart uint16 = 1
+)
+
+var (
+	MuxerErrorPIDNotFound      = errors.New("PID not found")
+	MuxerErrorPIDAlreadyExists = errors.New("PID already exists")
+	MuxerErrorPCRPIDInvalid    = errors.New("PCR PID invalid")
+)
+
+type Muxer struct {
+	ctx        context.Context
+	w          io.Writer
+	bitsWriter *astikit.BitsWriter
+
+	packetSize             int
+	tablesRetransmitPeriod int // period in PES packets
+
+	pm         programMap // pid -> programNumber
+	pmt        PMTData
+	nextPID    uint16
+	patVersion wrappingCounter
+	pmtVersion wrappingCounter
+
+	patBytes bytes.Buffer
+	pmtBytes bytes.Buffer
+
+	buf       bytes.Buffer
+	bufWriter *astikit.BitsWriter
+
+	esContexts              map[uint16]*esContext
+	tablesRetransmitCounter int
+}
+
+type esContext struct {
+	es *PMTElementaryStream
+	cc wrappingCounter
+}
+
+func newEsContext(es *PMTElementaryStream) *esContext {
+	return &esContext{
+		es: es,
+		cc: newWrappingCounter(0b1111), // CC is 4 bits
+	}
+}
+
+func MuxerOptionTablesRetransmitPeriod(newPeriod int) func(*Muxer) {
```

Comment:
Rename this function to `OptTablesRetransmitPeriod`

Reply:
How do we differentiate muxer and demuxer options after that?

Reply:
You're right. Then I think the best would be to rename this function `MuxerOptTablesRetransmitPeriod` and add a `Demuxer` prefix to demuxer options

Reply:
Done.

BTW, I'm quite positive we should also rename `astits.New` in demuxer.go to `NewDemuxer`. What do you think?

Reply:
You're right about `NewDemuxer`!

Reply:
Renamed